src/wallet/wallet.cpp

```diff
@@ -4396,6 +4400,9 @@ int CWallet::ScanForWalletTransactions(
                 if (consensus.NetworkUpgradeActive(pindex->pprev->nHeight,  Consensus::UPGRADE_SAPLING)) {
                     assert(pcoinsTip->GetSaplingAnchorAt(pindex->pprev->hashFinalSaplingRoot, frontiers.sapling));
                 }
+                if (consensus.NetworkUpgradeActive(pindex->pprev->nHeight,  Consensus::UPGRADE_NU5)) {
+                    assert(pcoinsTip->GetOrchardAnchorAt(pindex->pprev->hashFinalOrchardRoot, frontiers.orchard));
+                }
```

Comment:
In the case where the last block was before the Orchard activation height, should we assert that the last anchor is the empty root? (Like how we do in `ThreadNotifyWallets`